client.go

```diff
@@ -306,8 +305,58 @@ func (c *client) Read(ctx context.Context, fieldsToRead []string, entity DomainO
 // must contain values for all components of its primary key for the operation
 // to succeed. If `fieldsToRead` is provided, only a subset of fields will be
 // marshalled onto the given entities.
-func (c *client) MultiRead(context.Context, []string, ...DomainObject) (MultiResult, error) {
-	panic("not implemented")
+func (c *client) MultiRead(ctx context.Context, fieldsToRead []string, entities ...DomainObject) (MultiResult, error) {
+	if !c.initialized {
+		return nil, &ErrNotInitialized{}
+	}
+
+	if len(entities) == 0 {
+		return nil, fmt.Errorf("the number of entities to read is zero")
+	}
+
+	// lookup registered entity, registry will return error if registration
+	// is not found
+	var re *RegisteredEntity
+	var listFieldValues []map[string]FieldValue
+	for _, entity := range entities {
+		ere, err := c.registrar.Find(entity)
+		if err != nil {
+			return nil, err
+		}
+
+		if re == nil {
+			re = ere
+		} else if re != ere {
```

Comment:
For my own understanding, what are the different entity types?

Reply:
Basically this mean the multiread only support fetch one type of entity.